File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int a = 0;
        int b = 0;
        int n = nums.size();
        while(a < n){
            if(nums[a] != 0){
                swap(nums[a], nums[b]);
                b++;
            }
            a++;
        }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: list[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        temp = []
        for i in range(n):
            if nums[i] != 0:
                temp.append(nums[i])
        nz = len(temp)
        for i in range(nz):
            nums[i] = temp[i]
        for i in range(nz,n):
            nums[i] = 0


        